quizroyale/server/login_server.py

```diff
@@ -25,19 +28,30 @@ def login():
     except NoSuchUserException as e:
         return e.message, 400
 
-    return jsonify({'token': create_token(user.username, str(user.uid))},), 200
+    return jsonify(
+        {'token': create_token(user.username, str(user.uid),
+                               Secret.ACCESS_TOKEN_SECRET)}, ), 200
 
 
-def create_token(username: str, user_id: str):
-    return jwt.encode(
-        {
-            "sub": user_id,
-            "name": username,
-            "iat": int(datetime.datetime.utcnow().timestamp())  # issued at
-        },
-        'secret',
-        algorithm="HS256",
+@app.route('/renew_token', methods=['POST'])
+def validate_refresh_token():
+    # Do we need to check if user exist?
+    user = db.get_user_by_credentials(
+        request.form['username'],
+        request.form['password'],
     )
+    # Don't know how to store it in cookie at the moment
+    if user.refresh_token == request.form['refresh_token']:
+        decoded_request_token = jwt.decode(
+            request.form['refresh_token'],
+            Secret.REFRESH_TOKEN_SECRET,
+            "HS256",
+        )
+        if decoded_request_token['iat'] + datetime.timedelta(days=1) < int(
```

Comment:
I think date arithmetic should be abstracted to a separate component to make code cleaner